crates/tests/libs/registry/tests/volatile.rs

```diff
@@ -0,0 +1,53 @@
+use windows_registry::*;
+
+#[test]
+fn volatile() {
+    let test_key = "software\\windows-rs\\tests\\volatile";
+    _ = CURRENT_USER.remove_tree(test_key);
+
+    // Create a volatile key - this key will not persist when the system restarts
+    let key = CURRENT_USER
+        .options()
+        .create()
+        .volatile()
+        .write()
+        .open(test_key)
+        .unwrap();
+
+    // Write some data to the volatile key
+    key.set_u64("test_value", 12345u64).unwrap();
+
+    // Verify we can read the data
+    assert_eq!(key.get_u64("test_value").unwrap(), 12345u64);
+
+    // Verify the key was created (can be opened without create)
+    let read_key = CURRENT_USER.options().read().open(test_key).unwrap();
+
+    assert_eq!(read_key.get_u64("test_value").unwrap(), 12345u64);
+
+    // Clean up
+    _ = CURRENT_USER.remove_tree(test_key);
```

Comment:
The other tests don't bother with this since the test can fail early and so still leave this behind. So don't bother cleaning up.